src/Univ2LpOracle.sol

```diff
@@ -247,7 +249,26 @@ contract UNIV2LPOracle {
 
         // Get LP token supply
         uint256 supply = ERC20Like(src).totalSupply();
-        require(supply != 0, "UNIV2LPOracle/invalid-lp-token-supply");
+        require(supply > 0, "UNIV2LPOracle/invalid-lp-token-supply");
+
+        // Adjust LP token supply if fees are on
+        // We sacrifice some correctness for gas-savings by not querying param `feeTo`
+        // from the factory. This is only an issue during the time interval that Uniswap
+        // LP fees are toggled on/off and liquidity in the pair is added/removed.
+        // Uniswap Governance toggling fees is assumed to be a rare event, with liquidity
+        // for popular pairs constantly in flux. In the worst case the LP price would be off
+        // by ~0.05% which is insigificant relative to drift caused by the `hop` delay.
+        uint256 kLast = UniswapV2PairLike(src).kLast();
```

Comment:
Uniswap V2 does not normalize the reserve values to 18 decimals; since you're using a `k` computed from normalized reserves below to get `rootK`, you'll need to scale `kLast` by a factor of `10**(36 - dec0 - dec1)`.

Reply:
I totally see that `k` and `kLast` need to be scaled the same. However, it's unclear to me whether that means we should adjust `kLast` or just recompute `k` from non-normalized reserves. In terms of manipulation, it should be rather difficult to manipulate k and kLast.

Now I'm just rather confused in general though about the implementation of this fee calculation even works. You get drift between `k` and `kLast` whenever there a "swap" occurs after liquidity has either been added or removed from Uniswap. This drift (tho dynamic) is in place until the next individual adds/removes liquidity (thereby adjusting kLast). What does the relationship between these properties have to do with a fee? Doesn't really make sense to me. How is it that this converges to a 0.05% fee? Is the increased in the underlying assets of the LP tokens as swaps occur exactly proportional to the increase in the calculated supply adjustment (fee) due to drift in k from kLast?

Scenario:
Liquidity is added - kLast is set
Swap occurs - k goes up
Swap occurs - k goes up

Reply:
Like if you have someone add liquidity and no swap occurs. Then the (`rootK` > `rootKLast`) is false and no fee deduction is applied to the LP supply (and hence no change in LP price). That makes no sense. Whereas as soon as someone does a swap now there is a  fee deduction applied to LP supply (and hence LP price changes).

Reply:
I can type a more detailed reply later but maybe the following helps:

- `k` grows in response to trading fees
- you need to extract the protocol fee whenever someone adds or removes liquidity so that you fairly apply the dilution (think about `drip` calls in Maker)
- it makes complete sense to me that there's no fee withdrawn if no trades have occurred since the last assessment, since no fees have been collected
- you could add to the protocol fee on every trade instead, which would be more obvious, but that would be less gas efficient

Proving what % the fee taken this way actually represent requires a math calculation that I don't have time to do right now but I'll revisit later this evening.

Also, re this:

> I totally see that k and kLast need to be scaled the same. However, it's unclear to me whether that means we should adjust kLast or just recompute k from non-normalized reserves. In terms of manipulation, it should be rather difficult to manipulate k and kLast.

I guess you have your choice of how you solve it, but I suspect adjusting `kLast` might be slightly more gas efficient (although the difference should be extremely small).